openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -225,3 +156,275 @@ jint tcn_get_java_env(JNIEnv **env)
     }
     return JNI_OK;
 }
+
+// TODO: Share code with netty natives utilities.
+char* netty_internal_tcnative_util_prepend(const char* prefix, const char* str) {
+    if (prefix == NULL) {
```

Comment:
this method body should be equivalent to

    char *result = NULL;
    asprintf(&result, "%s%s", prefix ?: "", str);
    return result;

if you don't want to use GCC extensions, replace `prefix ?: ""` by `prefix ? prefix : ""`